Src/Base/AMReX_GpuAsyncArray.H

```diff
@@ -22,6 +22,19 @@ extern "C" {
 #endif
 
 namespace amrex {
+
+namespace type_traits {
+#if __cplusplus < 202002L
```

Comment:
Heh, GCC 10.1 identifies as C++17 if compiled with `-std=c++2a` but already adds the C++20 deprecation notice ;)